crates/router/src/services/api/client.rs

```diff
@@ -32,15 +35,59 @@ impl ProxyType {
     }
 }
 
+fn create_base_client(
+    proxy: Option<(ProxyType, String)>,
+) -> CustomResult<reqwest::Client, errors::ApiClientError> {
+    Ok(match proxy {
+        None => &PLAIN_CLIENT,
+        Some((ProxyType::Http, _)) => &HTTP_PROXY_CLIENT,
+        Some((ProxyType::Https, _)) => &HTTPS_PROXY_CLIENT,
+    }
+    .get_or_try_init(|| {
```

Comment:
By default reqwest library maintains a connection pool. Have you checked with traffic how many connections are we really opening or are they being reused. I am guessing Just clone would do the work and no need of onecell. Can you confirm this once?

Reply:
Yeah cloning would work since `reqwest::client` internally uses an Arc,

Using OnceCell to have the initial client reference & keep the client alive ,

The alternative could be that we have clients defined at the start of the application and pass them along with app state,

Using OnceCell for now to reduce overall changes in the code, If this works without any bugs we can extend it maybe


Lazy_CLIENT (static reference to client) -> Clone (reuse the client since it internally uses an Arc) -> use the cloned client for individual requests...

I'll get more insights into the  connections (I believe we also need to enable [tcp_keepalive](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.tcp_keepalive) for reusing connections).

For now the main purpose of this PR was to prevent any CPU overhead from creating new clients,
will address the system resource optimizations later

Reply:
> I'll get more insights into the connections (I believe we also need to enable [tcp_keepalive](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.tcp_keepalive) for reusing connections).
Cool. Let me know, We can have a comment added to code and this PR can be merged

Reply:
created an [issue](https://github.com/juspay/hyperswitch/issues/424) for this.